Fix handling of buckets with disabled bucket notifications

welansari opened the pull request:
- Added try/catch to _processObjectEntry to catch exception
and log them before returning an error
- When bucket notifications are disabled after being enabled,
the configuration is set to an empty object. Previously, this
case was not handled properly, leading to crashes as the
notification producer and processor incorrectly considered the
configuration available. This fix ensures proper handling of
empty configurations to prevent such issues.

Issue: BB-697

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-697 contains:


* `9.0.4.1`



Considering where you are trying to merge, I expected to find at least:



* `9.0.4.2`


Please check the `Fix Version/s` of BB-697, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

bert-e commented:
# Conflict




There is a conflict between your branch `bugfix/BB-697` and the
destination branch `hotfix/9.0.4`.

Please resolve the conflict on **the feature branch** (`bugfix/BB-697`).

```sh
git fetch && \
git checkout origin/bugfix/BB-697 && \
git merge origin/hotfix/9.0.4
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:bugfix/BB-697
```

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-697 contains:


* `9.0.12`

* `9.1.0`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `9.0.13`

* `9.1.0`


Please check the `Fix Version/s` of BB-697, or the target
branch of this pull request.

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

williamlardier commented on tests/unit/notification/NotificationQueuePopulator.js:
```diff
@@ -341,7 +341,7 @@ describe('NotificationQueuePopulator ::', () => {
             });
         });
-
+    
```
```suggestion

```

williamlardier commented on tests/unit/notification/NotificationQueuePopulator.js:
```diff
@@ -361,7 +361,7 @@ describe('NotificationQueuePopulator ::', () => {
             });
         });
-
+    
```
```suggestion

```

williamlardier commented on tests/unit/notification/NotificationQueuePopulator.js:
```diff
@@ -381,7 +381,7 @@ describe('NotificationQueuePopulator ::', () => {
             });
         });
-
+    
```
```suggestion

```

williamlardier commented on tests/unit/notification/NotificationQueuePopulator.js:
```diff
@@ -416,7 +416,7 @@ describe('NotificationQueuePopulator ::', () => {
             });
         });
-
+    
```
```suggestion

```

williamlardier commented on tests/unit/notification/NotificationQueuePopulator.js:
```diff
@@ -437,7 +437,7 @@ describe('NotificationQueuePopulator ::', () => {
             });
         });
-
+    
```
```suggestion

```

williamlardier commented on tests/unit/notification/NotificationQueuePopulator.js:
```diff
@@ -458,7 +458,7 @@ describe('NotificationQueuePopulator ::', () => {
             });
         });
-
+    
```
```suggestion

```

williamlardier commented on tests/unit/notification/NotificationQueuePopulator.js:
```diff
@@ -478,7 +478,7 @@ describe('NotificationQueuePopulator ::', () => {
         });
     });
-
+    
```
```suggestion

```

williamlardier commented on tests/unit/notification/NotificationQueuePopulator.js:
```diff
@@ -513,7 +513,7 @@ describe('NotificationQueuePopulator ::', () => {
         });
     });
-
+    
```
```suggestion

```

williamlardier commented on extensions/notification/NotificationQueuePopulator.js:
```diff
@@ -217,91 +217,88 @@ class NotificationQueuePopulator extends QueuePopulatorExtension {
-            };
-            this.log.debug('validating entry', {
+        } catch (error) {
+            const errorMessage = error?.toString?.() || JSON.stringify(error) || 'Unknown error';
```
This looks hacky, don't we have better ways to handle it? If that's an error it should always have a `.message`, no?

welansari replied:
yeah i think using `.message` should work fine in this case

Commit:
Fix error handling fallback in _processObjectEntry

Added try/catch to _processObjectEntry to catch exception
and log them before returning an error

Issue: BB-697

Commit:
Fix handling of disabled bucket notifications

When bucket notifications are disabled after being enabled,
the configuration is set to an empty object. Previously, this
case was not handled properly, leading to crashes as the
notification producer and processor incorrectly considered the
configuration available. This fix ensures proper handling of
empty configurations to prevent such issues.

Issue: BB-697

welansari commented:
/approve

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/BB-697` into
`development/9.0`
* [w/9.1/bugfix/BB-697](https://github.com/scality/backbeat/compare/development/9.1...w/9.1/bugfix/BB-697) will be merged into `development/9.1` 



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.6`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Build failed



The [build](https://github.com/scality/backbeat/actions?query=branch%3Aw/9.1/bugfix/BB-697) for [commit](https://github.com/scality/backbeat/commit/7e2ab99c42ee5235371fbd9e930d359736aa1edf) did not succeed in branch [w/9.1/bugfix/BB-697](https://github.com/scality/backbeat/compare/development/9.1...w/9.1/bugfix/BB-697)



*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/9.0`

* :heavy_check_mark: `development/9.1`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.6`



Please check the status of the associated issue BB-697.

Goodbye kerkesni.




*The following options are set:* **approve**
